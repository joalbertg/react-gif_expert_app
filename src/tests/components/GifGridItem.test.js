import React from 'react';
import { shallow } from 'enzyme';

import GifGridItem from '../../components/GifGridItem';

describe('Tests GifGridItem component', () => {
  const url = 'https://localhost:8080/something.gif';
  const title = 'something';

  let wrapper = shallow(<GifGridItem url={url} title={title} />);

  beforeEach(() => {
    wrapper = shallow(<GifGridItem url={url} title={title} />);
  });

  test('Should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Tests className attribute', () => {
    test('Should have card in className', () => {
      const text = wrapper.find('div').prop('className');
      expect(text.includes('card')).toBe(true);
    });

    test('Should have animate__animated and animate__fadeIn in className', () => {
      const text = wrapper.find('.card').prop('className');
      expect(text.includes('animate__animated')).toBe(true);
      expect(text.includes('animate__fadeIn')).toBe(true);
    });
  });

  describe('Tests url attribute', () => {
    const text = wrapper.find('.card img').prop('src');

    test('Should have a sub-string with https:// text', () => {
      expect(text.includes('https://')).toBe(true);
    });

    test('Should have a something with .gif', () => {
      expect(text.includes('.gif')).toBe(true);
    });

    test('Should have a alt with something text', () => {
      const text = wrapper.find('.card img').prop('alt');
      expect(text).toBe(title);
    });
  });

  test('Should have a title with something text', () => {
    const text = wrapper.find('.card p').text();
    expect(text).toBe(title);
  });
});


