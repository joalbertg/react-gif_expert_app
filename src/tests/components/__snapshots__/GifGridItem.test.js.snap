// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests GifGridItem component Should display correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="something"
    src="https://localhost:8080/something.gif"
  />
  <p>
    something
  </p>
</div>
`;
